main/src/ca/uwaterloo/flix/language/phase/Typer.scala

```diff
@@ -256,13 +256,36 @@ object Typer extends Phase[KindedAst.Root, TypedAst.Root] {
                 case Validation.Failure(errs) =>
                   val instanceErrs = errs.collect {
                     case UnificationError.NoMatchingInstance(tconstr) =>
-                      TypeError.NoMatchingInstance(tconstr.sym, tconstr.arg, tconstr.loc)
+                      if (tconstr.sym.name == "Eq")
+                        TypeError.MissingEq(tconstr.arg, tconstr.loc)
+                      else if (tconstr.sym.name == "Order")
+                        TypeError.MissingOrder(tconstr.arg, tconstr.loc)
```

Comment:
We should add at least one test for each of the new errors.

Reply:
OK, that is fair.